pilot/model/validation_test.go

```diff
@@ -2421,3 +2421,333 @@ func TestValidateHTTPRewrite(t *testing.T) {
+							LbPolicy: &routingv2.LoadBalancerSettings_Simple{
+								Simple: routingv2.LoadBalancerSettings_ROUND_ROBIN,
+							},
+						},
+						ConnectionPool: &routingv2.ConnectionPoolSettings{
+							Tcp:  &routingv2.ConnectionPoolSettings_TCPSettings{MaxConnections: 7},
+							Http: &routingv2.ConnectionPoolSettings_HTTPSettings{Http2MaxRequests: 11},
+						},
+						OutlierDetection: &routingv2.OutlierDetection{
+							Http: &routingv2.OutlierDetection_HTTPSettings{ConsecutiveErrors: 5},
+						},
+					},
+				},
+				{Name: "v2", Labels: map[string]string{"version": "v2"}},
+			},
+		}, valid: true},
+	}
+	for _, c := range cases {
+		if got := ValidateDestinationRule(c.in); (got == nil) != c.valid {
+			t.Errorf("ValidateDestinationRule failed on %v: got valid=%v but wanted valid=%v: %v",
+				c.name, got == nil, c.valid, got)
+		}
+	}
+}
+
+func TestValidateTrafficPolicyV2(t *testing.T) {
+	cases := []struct {
+		name  string
+		in    routingv2.TrafficPolicy
+		valid bool
+	}{
+		{name: "valid traffic policy", in: routingv2.TrafficPolicy{
+			LoadBalancer: &routingv2.LoadBalancerSettings{
+				LbPolicy: &routingv2.LoadBalancerSettings_Simple{
+					Simple: routingv2.LoadBalancerSettings_ROUND_ROBIN,
+				},
+			},
+			ConnectionPool: &routingv2.ConnectionPoolSettings{
+				Tcp:  &routingv2.ConnectionPoolSettings_TCPSettings{MaxConnections: 7},
+				Http: &routingv2.ConnectionPoolSettings_HTTPSettings{Http2MaxRequests: 11},
+			},
+			OutlierDetection: &routingv2.OutlierDetection{
+				Http: &routingv2.OutlierDetection_HTTPSettings{ConsecutiveErrors: 5},
+			},
+		},
+			valid: true},
+
+		{name: "invalid traffic policy, bad connection pool", in: routingv2.TrafficPolicy{
+			LoadBalancer: &routingv2.LoadBalancerSettings{
+				LbPolicy: &routingv2.LoadBalancerSettings_Simple{
+					Simple: routingv2.LoadBalancerSettings_ROUND_ROBIN,
+				},
+			},
+			ConnectionPool: &routingv2.ConnectionPoolSettings{},
+			OutlierDetection: &routingv2.OutlierDetection{
+				Http: &routingv2.OutlierDetection_HTTPSettings{ConsecutiveErrors: 5},
+			},
+		},
+			valid: false},
+
+		{name: "invalid traffic policy, bad outlier detection", in: routingv2.TrafficPolicy{
+			LoadBalancer: &routingv2.LoadBalancerSettings{
+				LbPolicy: &routingv2.LoadBalancerSettings_Simple{
+					Simple: routingv2.LoadBalancerSettings_ROUND_ROBIN,
+				},
+			},
+			ConnectionPool: &routingv2.ConnectionPoolSettings{
+				Tcp:  &routingv2.ConnectionPoolSettings_TCPSettings{MaxConnections: 7},
+				Http: &routingv2.ConnectionPoolSettings_HTTPSettings{Http2MaxRequests: 11},
+			},
+			OutlierDetection: &routingv2.OutlierDetection{},
+		},
+			valid: false},
+	}
+	for _, c := range cases {
+		if got := validateTrafficPolicy(&c.in); (got == nil) != c.valid {
+			t.Errorf("ValidateTrafficPolicy failed on %v: got valid=%v but wanted valid=%v: %v",
+				c.name, got == nil, c.valid, got)
+		}
+	}
+}
+
+func TestValidateConnectionPollV2(t *testing.T) {
```

Comment:
typo: pool. And you don't need v2 here do you? May be keep it for consistency purposes